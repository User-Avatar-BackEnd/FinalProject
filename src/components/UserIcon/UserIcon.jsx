import React from 'react';
import classNames from 'classnames';

import styles from './UserIcon.module.scss';

const UserIcon = ({ type, rank }) => {

  return (
    <div className={classNames(styles.UserIcon, styles[type], styles[rank])}>

    </div>
  );

}

export default UserIcon;