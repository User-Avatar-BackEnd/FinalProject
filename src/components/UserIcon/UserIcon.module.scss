$borderColor: #000;
$shadowColor: #ffffff;

.UserIcon {
  border-radius: 50%;
  padding: 5px;

  &.main {
    width: 250px;
    height: 250px;
    border: 2px solid $borderColor;
    box-shadow: inset 0 0 0 3px $shadowColor;
  }

  &.small {
    width: 30px;
    height: 30px;
    border: 1px solid $borderColor;
    box-shadow: inset 0 0 0 2px $shadowColor;
    transition: box-shadow .2s;
    cursor: pointer;

    &:hover {
      box-shadow: inset 0 0 0 2px orange;
    }

  }

  &.user {
    width: 15px;
    height: 15px;
    transition: box-shadow .2s;
  }


  &.NPC {
    background: url('../../images/rankIcons/npc-icon.jpg') 50% 50%/cover;
  }

  &.Crewman {
    background: url('../../images/rankIcons/crewman-icon.png') 50% 50%/cover;
  }

  &.Cossack {
    background: url('../../images/rankIcons/cossack-icon.jpg') 50% 50%/cover;
  }

  &.Centurion {
    background: url('../../images/rankIcons/centurion-icon.jpg') 50% 50%/cover;
  }

  &.Esaul {
    background: url('../../images/rankIcons/esaul-icon.jpg') 50% 50%/cover;
  }

  &.Ataman {
    background: url('../../images/rankIcons/ataman-icon.jpg') 50% 50%/cover;
  }

  &.Hetman {
    background: url('../../images/rankIcons/hetman-icon.jpg') 50% 50%/cover;
  }
}
