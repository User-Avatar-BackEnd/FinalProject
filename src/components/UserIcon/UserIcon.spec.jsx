import React from 'react';
import { shallow } from 'enzyme';

import UserIcon from './UserIcon';

describe('UserIcon component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<UserIcon />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});