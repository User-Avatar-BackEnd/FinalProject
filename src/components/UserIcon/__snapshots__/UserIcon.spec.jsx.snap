// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserIcon component should match snapshot 1`] = `
<div
  className="UserIcon undefined undefined"
/>
`;
